### PR TITLE
fix: use react-keybind for logs resize button & expand all logs accordions

### PR DESCRIPTION
**Pull request by Elessar1802**
# Description

Logs renderer uses useKeyDown. Replace that with react-keybind since that handles the case wherein if a input has focus it does not trigger the shortcut.

Also register a shortcut on key 'E' to expand/collapse all the accordions.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist

- [ ] The title of the PR states what changed and the related issues number (used for the release note).
- [ ] Does this PR require documentation updates?
- [ ] I've updated documentation as required by this PR.
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/Shared/Components/CICDHistory/History.components.tsx [hunk 1]
```diff
@@ -27,53 +28,57 @@ import { ReactComponent as ZoomIn } from '../../../Assets/Icon/ic-fullscreen.svg
...
+            pathname.includes('/logs') && (
+                <Tooltip
+                    placement="left"
+                    shortcutKeyCombo={{
+                        text: fullScreenView ? 'Exit fullscreen' : 'Enter fullscreen',
+                        combo: ['F'] as const,
+                    }}
+                >
+                    <div
+                        ref={zoomButtonRef}
```
Comment (AbhishekA1509) [inline src/Shared/Components/CICDHistory/History.components.tsx:66]: Can we make it button?

Comment (Elessar1802) [reply src/Shared/Components/CICDHistory/History.components.tsx:66]: made it into a button

**Review by AbhishekA1509 — COMMENTED** [review]
@vivek-devtron @Elessar1802  Should we instead of refs put the state in dependency itself at least of full screen?

**Comment by Elessar1802** [comment]
@AbhishekA1509 @vivek-devtron Completely removed the ref logic inside the shortcut handlers. Wrapped the handler in useCallback and put the handler as dependency of the useEffect that registers the shortcuts.

File: src/Shared/Components/CICDHistory/History.components.tsx [hunk 2]
```diff
@@ -27,53 +28,62 @@ import { ReactComponent as ZoomIn } from '../../../Assets/Icon/ic-fullscreen.svg
...
 
-export const LogResizeButton = ({ fullScreenView, setFullScreenView }: LogResizeButtonType): JSX.Element => {
-    const { pathname } = useLocation()
+export const LogResizeButton = withShortcut(
+    ({ fullScreenView, setFullScreenView, shortcut }: LogResizeButtonType & IWithShortcut): JSX.Element => {
+        const { pathname } = useLocation()
 
-    const keys = useKeyDown()
+        const toggleFullScreen = useCallback((): void => {
+            // NOTE: need to use ref due to the problem of stale function reference after registering the callback
```
Comment (AbhishekA1509) [inline src/Shared/Components/CICDHistory/History.components.tsx:36]: We can remove this comment

**Pull request merged**